### ISSUE TITLE
setField removes comment in patched line

### ISSUE THREAD

**Issue by christophlehmann** [networkteam/vignet#1]
Patch command

```json
        {
          "path": "projects/my/project/test.yaml",
          "setField": {
            "field": "spec.values.app.image.tag",
            "value": "6.2.1"
          }
        }
```

---

Other comments in that file are kept

**Comment by hlubek** [comment]
We should try to extract the `SetField` command and create a test case for this with some fixture YAMLs.

One way would be to unmarshal into https://pkg.go.dev/gopkg.in/yaml.v3#Node and work with the AST to set the the value (which is quite low-level) and handling cases of a key that does not exist or is not a "leaf".

We could also try to use https://pkg.go.dev/github.com/goccy/go-yaml instead.

**Comment by hlubek** [comment]
@florian-kuepper Would you like to give this a try?

**Comment by hlubek** [comment]
Fixed by 6b874fd
